pkg/loki/loki.go

```diff
@@ -221,100 +220,70 @@ func (c *Config) Clone() flagext.Registerer {
 // Validate the config and returns an error if the validation
 // doesn't pass
 func (c *Config) Validate() error {
+	var errs []error
+
 	if err := c.SchemaConfig.Validate(); err != nil {
-		return errors.Wrap(err, "invalid schema config")
+		errs = append(errs, errors.Wrap(err, "CONFIG ERROR: invalid schema config"))
 	}
 	if err := c.StorageConfig.Validate(); err != nil {
-		return errors.Wrap(err, "invalid storage config")
+		errs = append(errs, errors.Wrap(err, "CONFIG ERROR: invalid storage config"))
 	}
 	if err := c.QueryRange.Validate(); err != nil {
-		return errors.Wrap(err, "invalid queryrange config")
+		errs = append(errs, errors.Wrap(err, "CONFIG ERROR: invalid queryrange config"))
 	}
 	if err := c.Querier.Validate(); err != nil {
-		return errors.Wrap(err, "invalid querier config")
+		errs = append(errs, errors.Wrap(err, "CONFIG ERROR: invalid querier config"))
 	}
 	if err := c.QueryScheduler.Validate(); err != nil {
-		return errors.Wrap(err, "invalid query_scheduler config")
+		errs = append(errs, errors.Wrap(err, "CONFIG ERROR: invalid query_scheduler config"))
 	}
 	if err := c.TableManager.Validate(); err != nil {
-		return errors.Wrap(err, "invalid tablemanager config")
+		errs = append(errs, errors.Wrap(err, "CONFIG ERROR: invalid tablemanager config"))
 	}
 	if err := c.Ruler.Validate(); err != nil {
-		return errors.Wrap(err, "invalid ruler config")
+		errs = append(errs, errors.Wrap(err, "CONFIG ERROR: invalid ruler config"))
 	}
 	if err := c.Ingester.Validate(); err != nil {
-		return errors.Wrap(err, "invalid ingester config")
+		errs = append(errs, errors.Wrap(err, "CONFIG ERROR: invalid ingester config"))
 	}
 	if err := c.LimitsConfig.Validate(); err != nil {
-		return errors.Wrap(err, "invalid limits config")
+		errs = append(errs, errors.Wrap(err, "CONFIG ERROR: invalid limits config"))
 	}
 	if err := c.Worker.Validate(); err != nil {
-		return errors.Wrap(err, "invalid frontend-worker config")
+		errs = append(errs, errors.Wrap(err, "CONFIG ERROR: invalid frontend-worker config"))
 	}
 	if err := c.StorageConfig.BoltDBShipperConfig.Validate(); err != nil {
-		return errors.Wrap(err, "invalid boltdb-shipper config")
+		errs = append(errs, errors.Wrap(err, "CONFIG ERROR: invalid boltdb-shipper config"))
 	}
 	if err := c.IndexGateway.Validate(); err != nil {
-		return errors.Wrap(err, "invalid index_gateway config")
+		errs = append(errs, errors.Wrap(err, "CONFIG ERROR: invalid index_gateway config"))
 	}
 	if err := c.CompactorConfig.Validate(); err != nil {
-		return errors.Wrap(err, "invalid compactor config")
+		errs = append(errs, errors.Wrap(err, "CONFIG ERROR: invalid compactor config"))
 	}
 	if err := c.ChunkStoreConfig.Validate(); err != nil {
-		return errors.Wrap(err, "invalid chunk store config")
+		errs = append(errs, errors.Wrap(err, "CONFIG ERROR: invalid chunk store config"))
```

Comment:
nit so that the error message match the yaml section
```suggestion
		errs = append(errs, errors.Wrap(err, "CONFIG ERROR: invalid chunk_store_config config"))
```